src/main/java/com/capstone/producer/FlightInjectionHandler.java

```diff
@@ -33,21 +35,22 @@ public String handleFlightInjection(String receivedJSON) throws JsonProcessingEx
         if (jsonNode.isArray()) {
             for (JsonNode flight : jsonNode) {
                 try {
-                    var injectedFlight = objectMapper.convertValue(flight, FlightInfoFaid.class);
-
-                    var flightPosition = injectedFlight.getLastPosition();
-
-                    var flightInformationUpdate = new FlightInformation()
-                            .setFlightId("aero-" + injectedFlight.getFaFlightId())
-                            .setHeading(90) // TODO : replace this with real heading
-                            .setLocation(new GeographicCoordinates3D()
-                                    .setAltitude(flightPosition.getAltitude())
-                                    .setLatitude(flightPosition.getLatitude())
-                                    .setLongitude(flightPosition.getLongitude())
-                            ).setGroundSpeed(20000); // TODO : replace with real speed;
+                    FlightInfoFaid injectedFlight = objectMapper.convertValue(flight, FlightInfoFaid.class);
 
-
-                    KafkaProducer.emitFlightInformationUpdate(flightInformationUpdate);
+                    if(injectedFlight.getLastPosition() != null) {
+                        KafkaProducer.emitFlightInformationUpdate(handleRealInjection(injectedFlight));
+                    } else {
```

Comment:
This could be simplified by using an else-if. Would make it a little easier to read